ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/ViolationsByPolicyCategory.tsx

```diff
@@ -0,0 +1,150 @@
+    const widgetContainerResizeEntry = useResizeObserver(widgetContainer);
+    const { violationCounts, loading, error } = useViolationCounts('CATEGORY', ''); // TODO Implement query filtering
+
+    const countsBySeverity = getCountsBySeverity(violationCounts);
+    const bars = Object.entries(countsBySeverity).map(([severity, counts]) => {
+        const data = Object.entries(counts).map(([group, count]) => ({
+            name: severity,
+            x: group,
+            y: count,
+            label: `${severity}: ${count}`,
+        }));
+
+        return (
+            <ChartBar
+                barWidth={defaultChartBarWidth}
+                key={severity}
+                data={data}
+                labelComponent={<ChartTooltip constrainToVisibleArea />}
+                events={[
+                    navigateOnClickEvent(history, (props) => {
+                        const category = props?.datum?.xName as string;
+                        return linkForViolationsCategory(category);
+                    }),
+                ]}
+            />
+        );
+    });
+
+    let cardContent: ReactNode;
+
+    if (error) {
+        cardContent = (
+            <WidgetErrorEmptyState height={height} title="Unable to load data">
+                There was an error loading policy violation counts
+            </WidgetErrorEmptyState>
+        );
+    } else if (loading) {
+        cardContent = (
+            <Skeleton height={height} screenreaderText="Loading Policy Violations By Category" />
+        );
+    } else {
+        cardContent = (
+            <div className="pf-u-px-md" ref={setWidgetContainer} style={{ height }}>
+                <Chart
+                    ariaDesc="Number of violation by policy category, grouped by severity"
+                    ariaTitle="Policy Violations by Category"
+                    domainPadding={{ x: [30, 25] }}
+                    legendData={[
+                        { name: 'Low' },
+                        { name: 'Medium' },
+                        { name: 'High' },
+                        { name: 'Critical' },
+                    ]}
+                    legendPosition="bottom"
+                    height={chartHeight}
+                    width={widgetContainerResizeEntry?.contentRect.width} // Victory defaults to 450
+                    padding={{
+                        // TODO Auto-adjust padding based on screen size and/or max text length, if possible
+                        left: 180, // left padding is dependent on the length of the text on the left axis
+                        bottom: 75, // Adjusted to accommodate legend
+                    }}
+                    theme={patternflySeverityTheme}
+                >
+                    <ChartAxis
+                        tickLabelComponent={
+                            <LinkableChartLabel
+                                linkWith={({ text }) => linkForViolationsCategory(String(text))}
```

Comment:
If `linkWith` prop is a pure function, what do you think about factoring it out to module scope?

Integration tests have sometimes failed because element detached from the DOM, if inline callback function has different value on every render.